lib/cdo.rb

```diff
@@ -161,12 +161,21 @@ def curriculum_languages
       return @@curriculum_languages
     end
 
-    def curriculum_url(locale, path = '')
-      locale = locale.downcase.to_s
-      domain = "https://curriculum.code.org"
-      curriculum_languages.include?(locale) ?
-        File.join(domain, locale, path) :
-        File.join(domain, path)
+    def curriculum_url(locale, uri = '', autocomplete_partial_path = true)
+      if uri
+        uri = URI.parse(uri)
+
+        uri.host = "curriculum.code.org" if uri.host.nil? && autocomplete_partial_path
+        uri.scheme = "https" if uri.scheme.nil? && autocomplete_partial_path
+        uri.path = '/' + uri.path unless uri.path.start_with?('/')
```

Comment:
it's not clear to me why this is necessary; can you explain?

Reply:
uri.path requires an absolute path, so we need to have a leading `'/'` on them (for cases like `curriculum_url("es-mx", "test")`)

Reply:
ahhh, right. I forgot how relatively user-unfriendly `URI.parse` is :P